ignite/contrib/metrics/regression/canberra_metric.py

```diff
@@ -22,13 +25,21 @@ class CanberraMetric(_BaseRegression):
 
     """
 
+    def __init__(
+        self, output_transform: Callable = lambda x: x, device: Union[str, torch.device] = torch.device("cpu")
+    ):
+        self._sum_of_errors = None
+        super(CanberraMetric, self).__init__(output_transform, device)
+
+    @reinit__is_reduced
     def reset(self):
-        self._sum_of_errors = 0.0
+        self._sum_of_errors = torch.tensor(0.0, device=self._device)
 
     def _update(self, output):
         y_pred, y = output
         errors = torch.abs(y.view_as(y_pred) - y_pred) / (torch.abs(y_pred) + torch.abs(y.view_as(y_pred)))
```

Comment:
I'm wondering now why we used `view_as` ... Docstring states that both should be of the same shape.

Reply:
That is the current implementation. I can try to remove this.

Reply:
Yes, I know that it was here before. I'm just wondering now what was the good reason to do that.

Reply:
@vfdev-5 done, let's see.

Reply:
Seems like we used that `view_as` everywhere for other regression metrics. We have to provide a uniform approach in other PRs.